blocks/file-blocks/markdown-edit/extensions.ts

```diff
@@ -124,6 +126,31 @@ export function makeExtensions({
       }
     }),
 
+    EditorView.domEventHandlers({
+      paste(e, view) {
+        const value = e.clipboardData?.items[0];
+        if (value.type === "image/png") {
+          const file = value.getAsFile();
+          if (file) {
+            const reader = new FileReader();
+            reader.onload = (e) => {
+              const image = e.target?.result;
+              if (image) {
+                view.dispatch({
+                  changes: {
+                    from: view.state.selection.main.from,
+                    to: view.state.selection.main.to,
+                    insert: `![${file.name}](${image})`,
+                  },
+                });
+              }
+            };
+            reader.readAsDataURL(file);
```

Comment:
for robustness you could put some reasonable size limit on pasted files, although it's not super clear what "reasonable" is here.

https://stackoverflow.com/questions/695151/data-protocol-url-size-limitations

Reply:
good call!